src/CLI/clitree/cli-xml/ipv4.xml

```diff
@@ -24,109 +24,102 @@ limitations under the License.
     xsi:schemaLocation="http://www.dellemc.com/sonic/XMLSchema
                         http://www.dellemc.com/sonic/XMLSchema/clish.xsd">
 
-
-
 <!--=======================================================-->
-<!--              show ip arp interface                    -->
+<!--              show ip arp changes                      -->
 <!--=======================================================-->
 <PTYPE
-    name="INTF_NAME"
+    name="INTF_TYPE"
     method="select"
     pattern="Ethernet Vlan PortChannel Management(eth)"
     help="Interface type"/>
 
+
 <VIEW name="enable-view">
-<!--
-    <COMMAND
-            name="clear ip"
-            help="Clear ip command">
-    </COMMAND>
-    <COMMAND
-            name="debug ip"
-            help="Debug ip command">
-    </COMMAND>
--->
     <COMMAND
         help="show IP commands"
         name="show ip"
         >
-        <ACTION builtin="clish_nop"/>
+    <ACTION builtin="clish_nop"/>
     </COMMAND>
 
+<!-- show ip arp <ip-addr> commands -->
     <COMMAND
-        name="show ip arp"
-        help="show IP ARP commands">
+        help="show ARP commands"
+        name="show ip arp">
         <PARAM
-            name="interface"
-            help="Specify interface type"
+            name="arp-subcommands"
+            help="ARP subcommands"
+            mode="switch"
             ptype="SUBCOMMAND"
-            mode="subcommand"
-            optional="false">
+            optional="true">
+                <PARAM
+                    help="ARP entry for this IP address"
+                    name="ip-addr"
+                    ptype="IP_ADDR"
+                    optional="true">
+                </PARAM>
+        </PARAM>
+       <ACTION>
+           if test "${arp-subcommands}" = "ip-addr"; then&#xA;
+               python $SONIC_CLI_ROOT/sonic-cli-arp-neigh.py get_sonic_neigh_sonic_neigh_neigh_table arp_show.j2 IPv4 ${ip-addr}&#xA;
+           else&#xA;
```

Comment:
We can simply combine these 2 and pass ${ip-addr} in both cases. It will either be empty or will have a value. The python script can handle 2 cases.

Reply:
Done!